cmake/CubCudaConfig.cmake

```diff
@@ -128,6 +130,17 @@ endif()
 
 message(STATUS ${arch_message})
 
+# Define CUB_PTX_TARGETS to restrict the number of kernel tunings that are
+# instantiated. This is distinct from the logic in cub-config.cmake, since
+# CUB detects PTX arches after finding the CUB package.
+# After 11.5, __CUDA_ARCH_LIST__ should be available, and CUB will automatically
+# use that instead.
+if (CMAKE_CUDA_COMPILER_ID STREQUAL "NVIDIA" AND
+    CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.5)
+  list(JOIN ptx_targets "\\," ptx_targets)
+  target_compile_definitions(_CUB_CUB INTERFACE "CUB_PTX_TARGETS=${ptx_targets}")
```

Comment:
I don't think `_CUB_CUB` is defined here when building CUB as a standalone project:
```bash
stderr 1: CMake Error at cmake/CubCudaConfig.cmake:141 (target_compile_definitions):
  Cannot specify compile definitions for target "_CUB_CUB" which is not built
  by this project.
```

Reply:
We should just prevent CUB from being used standalone at some point. Thrust + CUB submodule is preferred, especially for dev builds that would be hitting this code path.

But in the meantime, good catch! I'll fix this.

Reply:
Changed it to `cub.compiler_interface`.